engine/swarm/services.md

```diff
@@ -264,24 +264,50 @@ Swarm mode lets you network services in a couple of ways:
 * publish ports externally to the swarm using ingress networking
 * connect services and tasks within the swarm using overlay networks
 
-### Publish ports externally to the swarm
+### Publish ports
 
-You publish service ports externally to the swarm using the `--publish
-<TARGET-PORT>:<SERVICE-PORT>` flag. When you publish a service port, the swarm
-makes the service accessible at the target port on every node regardless if
-there is a task for the service running on the node.
+When you create a swarm service, you can publish that service's ports to hosts
+outside the swarm in two ways:
 
-For example, imagine you want to deploy a 3-replica nginx service to a 10-node
-swarm as follows:
+- [You can rely on the routing mesh](#publish-a services-ports-using-the-routing-mesh).
+  When you publish a service port, the swarm makes the service accessible at the
+  target port on every node, regardless of whether there is a task for the
+  service running on that node or not. This is less complex but creates some
+  additional networking overhead.
+
+- [You can publish a service task's port directly on the swarm node](#publish-a-services-ports-directly-on-the-swarm-node)
+  where that service is running. This feature is available in Docker 1.13 and
+  higher. This creates less networking overhead by passing the routing mesh, but
+  you are responsible for routing requests to your service's nodes, and you can
+  only run one task for a given service on a given swarm node.
```

Comment:
This restriction is only in place if you specify the published port. If you don't specify it, an ephemeral port will be allocated on the node.